File: frontend/src/types/competitor.ts
export interface Competitor {
  id: string;
  user_id: string;
  name: string;
  website?: string;
  category?: string;
  region?: string;
  rating?: number;
  notes?: string;
  created_at: string;
}

export interface CompetitorCreate {
  name: string;
  website?: string;
  category?: string;
  region?: string;
  rating?: number;
  notes?: string;
}

export interface CompetitorAnalysisSummary {
  total_competitors: number;
  average_rating?: number;
  top_competitor?: string;
  categories: string[];
}

File: backend/app/models/competitor.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime
import uuid


class CompetitorBase(BaseModel):
    name: str
    website: Optional[str] = None
    category: Optional[str] = None
    region: Optional[str] = None
    rating: Optional[float] = None
    notes: Optional[str] = None


class CompetitorCreate(CompetitorBase):
    pass


class CompetitorUpdate(BaseModel):
    name: Optional[str] = None
    website: Optional[str] = None
    category: Optional[str] = None
    region: Optional[str] = None
    rating: Optional[float] = None
    notes: Optional[str] = None


class CompetitorOut(CompetitorBase):
    id: uuid.UUID
    user_id: uuid.UUID
    created_at: datetime

    class Config:
        from_attributes = True


class CompetitorAnalysisSummary(BaseModel):
    total_competitors: int
    average_rating: Optional[float]
    top_competitor: Optional[str]
    categories: list[str]
